ribs/emitters/_genetic_algorithm_emitter.py

```diff
@@ -0,0 +1,183 @@
+            check_batch_shape(self._initial_solutions, "initial_solutions",
+                              archive.solution_dim, "archive.solution_dim")
+
+        EmitterBase.__init__(
+            self,
+            archive,
+            solution_dim=archive.solution_dim,
+            bounds=bounds,
+        )
+        if self._os == 'isoline':
+            self._operator = _get_op(os)(lower_bounds=self._lower_bounds,
+                                         upper_bounds=self._upper_bounds,
+                                         seed=self._seed,
+                                         iso_sigma=self._iso_sigma,
+                                         line_sigma=self._line_sigma)
+        elif self._os == 'gaussian':
+            self._operator = _get_op(os)(lower_bounds=self._lower_bounds,
+                                         upper_bounds=self._upper_bounds,
+                                         seed=self._seed,
+                                         sigma=self._sigma)
+        else:
+            raise ValueError(f"{self._os} not a supported operator.")
+
+    @property
+    def x0(self):
+        """numpy.ndarray: Initial Solution (if initial_solutions is not
+        set)."""
+        return self._x0
+
+    @property
+    def iso_sigma(self):
+        """float: Scale factor for the isotropic distribution used to
+        generate solutions when the archive is not empty."""
+        return self._iso_sigma
+
+    @property
+    def line_sigma(self):
+        """float: Scale factor for the line distribution used when generating
+        solutions."""
+        return self._line_sigma
+
+    @property
+    def sigma(self):
+        """float or numpy.ndarray: Standard deviation of the (diagonal) Gaussian
+        distribution when the archive is not empty."""
+        return self._sigma
+
+    @property
+    def initial_solutions(self):
+        """numpy.ndarray: The initial solutions which are returned when the
+        archive is empty (if x0 is not set)."""
+        return self._initial_solutions
+
+    @property
+    def batch_size(self):
+        """int: Number of solutions to return in :meth:`ask`."""
+        return self._batch_size
+
+    def ask(self):
+        """Creates solutions with operator provided.
+
+
+        If the archive is empty and ``self._initial_solutions`` is set, we
+        return ``self._initial_solutions``. If ``self._initial_solutions`` is
+        not set and the archive is still empty, we operate on the initial
+        solution (x0) provided. Otherwise, we sample parents from the archive
+        to be used as input to our operator
+
+        Returns:
+            If the archive is not empty, ``(batch_size, solution_dim)`` array
+            -- contains ``batch_size`` new solutions to evaluate. If the
+            archive is empty, we return ``self._initial_solutions``, which
+            might not have ``batch_size`` solutions.
+        """
+
+        if self._os == 'isoline':
```

Comment:
The operator is not exposed to the archive, so the sampling logic will always be there, albeit operator.parent_type makes it cleaner. 

The different ask behavior for isoline and gaussian on an empty archive will also require separate logic.

To my understanding the only way we can clean it up is by granting archive and initial_solution access to the operator, so the logic separates in operator ask functions rather than this emitter.